SparkPost/2web2ui
# Upgrade to React 16.8

# What I Did
 - Bumped react and react-dom to 16.8 (16.8.2 at the moment)
 - Installed and configured the `react-hooks` ESlint plugin to protect us from tripping over the [React Hooks Rules](https://reactjs.org/docs/hooks-rules.html)

# How To Test
 - Check dev, test and builds all function
    - `npm start`
    - `npm test`
    - `npm run build` (see docs/perf-testing.md for help on serving a build locally)
 - Use a hook ([docs here](https://reactjs.org/docs/hooks-intro.html))

# To Do
 - [x] Rebase to `master` after `greenkeeper/initial` goes down.
